Client/Frontend/Home/ActivityStreamTopSitesCell.swift

```diff
@@ -57,17 +57,30 @@ class TopSiteItemCell: UICollectionViewCell {
 
         contentView.layer.cornerRadius = TopSiteCellUX.CellCornerRadius
         contentView.layer.masksToBounds = true
-
         contentView.addSubview(titleLabel)
         contentView.addSubview(imageView)
         contentView.addSubview(selectedOverlay)
+    }
 
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        contentView.backgroundColor = UIColor.lightGrayColor()
+        imageView.image = nil
+        titleLabel.text = ""
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
         let heightInset = frame.height * TopSiteCellUX.TitleInsetPercent
-        titleLabel.snp_makeConstraints { make in
-            make.edges.equalTo(self).inset(UIEdgeInsets(top: heightInset, left: 0, bottom: 0, right: 0))
-        }
 
-        imageView.snp_makeConstraints { make in
+        titleLabel.snp_remakeConstraints { make in
+            make.edges.equalTo(contentView).inset(UIEdgeInsets(top: heightInset, left: 0, bottom: 0, right: 0))
+        }
+        imageView.snp_remakeConstraints { make in
```

Comment:
Not sure if this was in `layoutSubviews` before but you should avoid mixing autolayout code with `layoutSubviews` calls or things can get a bit messy/weird. If you're looking for a frequent way to update the constraints, how about `updateConstraints`?